token/js/src/extensions/transferHook/state.ts

```diff
@@ -49,3 +52,64 @@ export function getTransferHookAccount(account: Account): TransferHookAccount |
         return null;
     }
 }
+
+/** ExtraAccountsMeta as stored by the transfer hook program */
+export interface ExtraAccountsMeta {
+    discriminator: number;
+    addressConfig: Uint8Array;
+    isSigner: boolean;
+    isWritable: boolean;
+}
+
+/** Buffer layout for de/serializing an ExtraAccountsMeta */
+export const ExtraAccountsMetaLayout = struct<ExtraAccountsMeta>([
+    u8('discriminator'),
+    blob(32, 'addressConfig'),
+    bool('isSigner'),
+    bool('isWritable'),
+]);
+
+export interface ExtraAccountsMetaList {
+    count: number;
+    extraAccounts: ExtraAccountsMeta[];
+}
+
+/** Buffer layout for de/serializing a list of ExtraAccountsMeta prefixed by a u32 length */
+export const ExtraAccountsMetaListLayout = struct<ExtraAccountsMetaList>([
+    u32('count'),
+    seq<ExtraAccountsMeta>(ExtraAccountsMetaLayout, greedy(ExtraAccountsMetaLayout.span), 'extraAccounts'),
+]);
+
+/** Unpack an extra account metas account and parse the data into a list of ExtraAccountMetas */
+export function getExtraAccountMetas(account: AccountInfo<Buffer> | null): ExtraAccountsMeta[] | null {
```

Comment:
You have a null check in `addExtraAccountsToInstruction`, do you need this parameter to be nullable? I think it makes sense for it to be a valid `AccountInfo` if you are attempting to deserialize it